src/coreclr/vm/class.cpp

```diff
@@ -661,6 +661,17 @@ HRESULT EEClass::AddMethodDesc(
     PCCOR_SIGNATURE sig;
     if (FAILED(hr = pImport->GetSigOfMethodDef(methodDef, &sigLen, &sig)))
         return hr;
+
+    SigParser sigParser(sig, sigLen);
+    ULONG offsetOfAsyncDetails;
+    bool isValueTask;
+    MethodReturnKind returnKind = ClassifyMethodReturnKind(sigParser, pModule, &offsetOfAsyncDetails, &isValueTask);
+    if (returnKind != MethodReturnKind::NormalMethod)
```

Comment:
Should this have TODO Async? 

This will need to be fixed before `FEATURE_RUNTIME_ASYNC` is enabled in dotnet/runtime.

Reply:
I am going to log an issue on EnC implementation with Async, but can have TODOs as well.

Reply:
I've added TODOs and here is a tracking issue on EnC: https://github.com/dotnet/runtime/issues/115095